besu/src/main/java/org/hyperledger/besu/cli/subcommands/blocks/BlocksSubCommand.java

```diff
@@ -150,37 +159,48 @@ public void run() {
     @Override
     public void run() {
       parentCommand.parentCommand.configureLogging(false);
-      LOG.info("Import {} block data from {}", format, blocksImportFile);
 
       checkCommand(parentCommand);
       checkNotNull(parentCommand.rlpBlockImporter);
       checkNotNull(parentCommand.jsonBlockImporterFactory);
-
+      if (blockImportFileOption.isEmpty()) {
+        throw new ParameterException(spec.commandLine(), "No files specified to import.");
+      }
+      LOG.info("Import {} block data from {} files", format, blockImportFiles.size());
       final Optional<MetricsService> metricsService = initMetrics(parentCommand);
 
-      try {
-        // As blocksImportFile even if initialized as null is injected by PicoCLI and param is
-        // mandatory. So we are sure it's always not null, we can remove the warning.
-        //noinspection ConstantConditions
-        final Path path = blocksImportFile.toPath();
-        final BesuController<?> controller = createController();
-        switch (format) {
-          case RLP:
-            importRlpBlocks(controller, path);
-            break;
-          case JSON:
-            importJsonBlocks(controller, path);
-            break;
-          default:
-            throw new ParameterException(
-                spec.commandLine(), "Unsupported format: " + format.toString());
+      try (final BesuController<?> controller = createController()) {
+        for (final Path path : blockImportFiles) {
+          try {
+            LOG.info("Importing from {}", path);
+
+            switch (format) {
+              case RLP:
+                importRlpBlocks(controller, path);
+                break;
+              case JSON:
+                importJsonBlocks(controller, path);
+                break;
+              default:
+                throw new ParameterException(
+                    spec.commandLine(), "Unsupported format: " + format.toString());
+            }
+          } catch (final FileNotFoundException e) {
+            if (blockImportFiles.size() == 1) {
+              throw new ExecutionException(
+                  spec.commandLine(), "Could not find file to import: " + path);
+            } else {
+              LOG.error("Could not find file to import: {}", path);
+            }
```

Comment:
So we continue on if one of the files isn't found? *If* the typical use case is to supply block files that have contiguous blocks, as a user I would prefer a bail on the first indication of trouble.

Reply:
This code represents what other clients  They only halt if one file is presented and plow through if multiple files are presented (but share every error).  It's a feature we need to run this on hive efficiently.

One use case is one set represents a consensus fork and the next set has overlapping block numbers, but different and correct blocks.  Or if one file would up orphaned by the end and the next file contains the live chain head.

Reply:
Ah, ok. So we need to continue on in the case where there's an import failure but what about the FileNotFoundException? Do we need to have that behavior as well in order to conform?

Reply:
Yes. An error will be logged, user error could be a buggy auto-generated scripts that used to work.